components/widgets/datetime/index.js

```diff
@@ -24,7 +24,12 @@ export default class DateTime extends Component {
 
   componentDidMount () {
     const { interval } = this.props
-    this.timeout = setTimeout(() => this.setState({ date: new Date() }), interval)
+    this.timeout = setTimeout(() => {
+      this.setState({
+        date: new Date()
+      })
+      this.componentDidMount()
+    }, interval)
   }
```

Comment:
I have created a small example, please have a look at it: https://codesandbox.io/s/q4l2q6yr76

Reply:
The reason why I wanted to stick to settimeout was to keep it aligned with the other modules. If you prefer the setinterval approach, then it is fine by me.

Reply:
For the other components, the update handling is a bit different than here. We can use the example from the comment above as a reference.

Reply:
Changed according to the example